src/Core/Framework/MessageQueue/ScheduledTask/Scheduler/TaskScheduler.php

```diff
@@ -121,9 +121,8 @@ private function buildCriteriaForAllScheduledTask(): Criteria
     private function queueTask(ScheduledTaskEntity $taskEntity): void
     {
         $taskClass = $taskEntity->getScheduledTaskClass();
-        $taskClassParents = class_parents($taskClass);
 
-        if ($taskClassParents === false || !\in_array(ScheduledTask::class, $taskClassParents, true)) {
+        if (!\is_a($taskClass, ScheduledTask::class, true)) {
```

Comment:
due to this change, Psalm can now determine which object type the `$task` variable will have. 

that is why the annotation from line 132 is unnecessary

Reply:
Oh wow, didn’t expect that level of smartness 😅. Will change 👍